rust/foxglove/src/websocket/ws_protocol/server/server_info.rs

```diff
@@ -6,6 +6,25 @@ use serde::{Deserialize, Serialize};
 
 use crate::websocket::ws_protocol::JsonMessage;
 
+/// Type for serializing timestamps in the server info message.
+///
+/// This exists to allow for sending timestamps that are represented as
+/// u64 in memory over JSON without loss of precision.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct SerializedTimestamp {
+    pub sec: u32,
+    pub nsec: u32,
+}
+
+impl SerializedTimestamp {
+    fn from_nsecs(timestamp: u64) -> Self {
+        SerializedTimestamp {
+            sec: (timestamp / 1e9 as u64) as u32,
```

Comment:
There's a chance of overflow here, because `u64::MAX/1e9` requires 34 bits. It'd be better to saturate instead truncating:


```suggestion
            sec: u32::try_from(timestamp / 1e9 as u64).unwrap_or(u32::MAX),
```